src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets

```diff
@@ -209,17 +219,59 @@ Copyright (c) .NET Foundation. All rights reserved.
        * Assets from the referenced packages. These will be implicitly included when nuget
          restores the package and includes the package props file for the package.
   -->
+
+    <!-- StaticWebAssets from the current project come from ResolveCurrentProjectStaticWebAssetsInputs which is a dependency of this target. -->
+    <!-- StaticWebAssets from referenced projects come from ResolveReferencedProjectsStaticWebAssets which is a dependency of this target. -->
+    <!-- StaticWebAssets from packages are already available, so we don't do anything. -->
   <Target
     Name="ResolveStaticWebAssetsInputs"
-    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)">
+    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)" />
+
+  <!-- This is a helper task to compute the project references we need to invoke to retrieve
+       the static assets for a given application. We do it this way so that we can
+       pass additional build properties to compute the assets from the package when referenced
+       as a project. For example, Identity uses this hook to extend the project reference and
+       pass in the bootstrap version to use.
+  -->
 
-    <!-- StaticWebAssets from the current project -->
+  <Target Name="ResolveReferencedProjectsStaticWebAssets" DependsOnTargets="$(ResolveReferencedProjectsStaticWebAssetsDependsOn)">
 
-    <PropertyGroup>
-      <StaticWebAssetBasePath Condition="$(StaticWebAssetBasePath) == ''">_content/$(PackageId)</StaticWebAssetBasePath>
-    </PropertyGroup>
+    <ItemGroup>
+      <_StaticWebAssetsProjectReference Include="@(AnnotatedProjects)" />
+    </ItemGroup>
+
+    <MSBuild
+      Condition="'@(_StaticWebAssetsProjectReference->Count())' != '0' and '%(_StaticWebAssetsProjectReference.BuildReference)' == 'true' and '@(ProjectReferenceWithConfiguration)' != ''"
+      Targets="GetCurrentProjectStaticWebAssets"
+      Properties="%(_StaticWebAssetsProjectReference.SetConfiguration); %(_StaticWebAssetsProjectReference.SetPlatform);  %(_StaticWebAssetsProjectReference.SetTargetFramework)"
+      RemoveProperties="%(_StaticWebAssetsProjectReference.GlobalPropertiesToRemove)"
+      Projects="@(_StaticWebAssetsProjectReference)"
+      BuildInParallel="$(BuildInParallel)"
+      ContinueOnError="!$(BuildingProject)"
+      SkipNonexistentTargets="true">
+
+      <Output TaskParameter="TargetOutputs" ItemName="_ReferencedProjectStaticWebAssets" />
+    </MSBuild>
 
     <ItemGroup>
+      <StaticWebAsset Include="%(_ReferencedProjectStaticWebAssets.Identity)">
+        <ContentRoot>%(ContentRoot)</ContentRoot>
+        <BasePath>%(BasePath)</BasePath>
```

Comment:
What's the advantage of adding identity rather than just using the item that came out? Are you trying to filter metadata?  If so, does [`KeepMetadata`](https://docs.microsoft.com/en-us/visualstudio/msbuild/item-element-msbuild?view=vs-2019) help?

Reply:
Didn't know about keep metadata, switched it to use that.